tool.go

```diff
@@ -319,11 +319,13 @@ func main() {
 					}
 
 					for _, decl := range testPkg.Archive.Declarations {
-						if strings.HasPrefix(decl.FullName, testPkg.ImportPath+".Test") {
+						switch {
+						case decl.FullName == testPkg.ImportPath+".TestMain":
```

Comment:
Confirmed, if the signature is `func TestMain(t *testing.T)` then the `go` tool treats it like a normal test:

```
31 $ go test -v
=== RUN   TestMain
--- PASS: TestMain (0.00s)
=== RUN   Test1
--- PASS: Test1 (0.00s)
PASS
ok  	github.com/shurcooL/play/31	0.022s
```